pkg/controllers/sync/dispatch/retain.go

```diff
@@ -96,6 +96,25 @@ func RetainOrMergeClusterFields(
 	return nil
 }
 
+func AddAnnotationToRecordLabelsAndAnnotations(obj *unstructured.Unstructured) error {
+	// Record the annotation/label keys in the template so we can diff it against the cluster object during retention
+	// to determine whether an annotation/label has been deleted from the template.
+	if _, err := annotationutil.AddAnnotation(
```

Comment:
We can avoid having to return an error by using `obj.GetAnnotations` and `obj.SetAnnotations` directly.